src/app_unit.rs

```diff
@@ -289,6 +289,92 @@ impl Au {
     }
 }
 
+#[cfg(feature = "num_traits")]
+impl num_traits::ToPrimitive for Au {
+    // Float conversions
+
+    #[inline]
+    fn to_f32(&self) -> Option<f32> {
+        Some(self.to_f32_px())
+    }
+
+    #[inline]
+    fn to_f64(&self) -> Option<f64> {
+        Some(self.to_f64_px())
+    }
+
+    // Signed int conversions
+
+    #[inline]
+    fn to_i8(&self) -> Option<i8> {
+        None
+    }
+
+    #[inline]
+    fn to_i16(&self) -> Option<i16> {
+        None
+    }
+
+    #[inline]
+    fn to_i32(&self) -> Option<i32> {
+        Some(self.to_px())
+    }
+
+    #[inline]
+    fn to_i64(&self) -> Option<i64> {
+        Some(self.to_px() as i64)
```

Comment:
I think you can just implement to to_i64() and the others have default implementations.

Reply:
It might be worth special casing `i32`, because that's what we actually have? I guess casting to i64 and back again is probably pretty cheap / likely to be inlined. But still...